src/equisolve/numpy/scripts/equi_kit.py

```diff
@@ -0,0 +1,299 @@
+from abc import ABCMeta, abstractmethod
+from copy import deepcopy
+
+from typing import List, Set, Optional, Union, Tuple, TypeVar
+
+from equisolve.numpy.models.linear_model import Ridge
+
+from rascaline import Composition, LodeSphericalExpansion, SphericalExpansion
+
+from rascaline import Composition, SoapRadialSpectrum, SoapPowerSpectrum
+
+from equistore import TensorBlock, TensorMap
+from equistore.operations import join, ones_like, slice, sum_over_samples, mean_over_samples
+
+# TODO from utils.models.soap import compute_power_spectrum
+
+import ase
+
+import numpy as np
+
+# Workaround for typing Self with inheritance for python <3.11
+# see https://peps.python.org/pep-0673/
+TEquiKitScript= TypeVar("TEquiKitScript", bound="EquiKitScript")
+
+class EquiKitScript(metaclass=ABCMeta):
+    """
+    An EquiScript is a merge of a representation calculator and a ML model.
+
+    EquiKitScript supports scikit-learn like transformers and estimators that
+    have a fit and predict function and can handle equistore objects.
+    For some applications (e.g. long-range interaction, multi-spectra) we want to apply transformers
+    seperately on different representations (e.g. because they are on different scales)
+    and join them together for the estimation.
+
+    Workflow
+    ```python
+        hypers = ...
+        script = EquiKitScript(hypers, parameter_keys=["values", "positions"])
+        Xi = script.compute(frames) # output is (X0, ... XN)
+        y = ase_to_tensormap(frames, energy="energy", forces="forces")
+
+        script.fit(Xi, y, **estimator_fit_kwargs)
+        # fit logic:
+        #  transformer_X[0].transform(X0), ..., transformer_X[N].transform(XN) | transformer_y.transform(y)
+        #               X0_t,              ...,              XN_t              |            y_t
+        #
+        #  join(X0_t, ..., XN_t)
+        #           X_jt
+        #
+        #  estimator.fit(X_jt, y_t)
+
+        script.forward(Xi)
+    ```
+
+    COMMENT class name is too general, can we distinguish it to Hamiltonian learning and CG-NN applications where postprocessing happens after the join to compute errors?
+            Might be not important because these applications require gradient optimization, so they would be in any not within the Scikit scope
+
+    COMMENT The logic is a bit unnecessary entangled with atomistic data, but I would not bother with it for now
+    """
+    def __init__(self, hypers, *, feature_aggregation="mean", transformer_X=None, transformer_y=None, estimator=None):
```

Comment:
I think `hypers` should be dictionary with the key being the calculator name and the value the hypers. This this we are tightly bound to rascaline but this is fine for me!

And do I get this correctly that I can give a transformer `transformer_X` and `transformer_y` and the class will do this automatically? This is super nice!

Reply:
> I think hypers should be dictionary with the key being the calculator name and the value the hypers. This this we are tightly bound to rascaline but this is fine for me!

that is the case at the moment (sorry nowhere documented, wanted to give example, but because of the issue with the equistore libraries import I did not manage

> And do I get this correctly that I can give a transformer transformer_X and transformer_y and the class will do this automatically? This is super nice!

in the `fit` and `forward` function it is done